static/js/components/ObjPageAnnotations.jsx

```diff
@@ -112,6 +122,38 @@ const ObjPageAnnotations = ({ annotations }) => {
     },
   ];
 
+  // if there are any spectra given, we should add the spectrum annotations after object annotations
+  if (spectra.length > 0) {
```

Comment:
Is it possible for `spectra` to ever be passed in as `null` or `undefined` while the data are still being fetched? If so we need to be able to handle that here gracefully (this will error as-is if so).

Reply:
Still wondering this

Reply:
Yeah sorry I think we can just add a couple more conditions to that if, that's not a problem.

Reply:
like `if (spectra && spectra.length)`